packages/config/src/utils.ts

```diff
@@ -0,0 +1,32 @@
+import type {
+  ConfigSidebarFolder,
+  ConfigSidebarLink,
+  ConfigSidebarPage,
+  SidebarItem,
+} from '@/configTypes'
+import { Value } from '@sinclair/typebox/value'
+
+/** Checks an object against a Typebox schema and returns the errors */
+export function getSchemaErrors<
+  T extends ConfigSidebarFolder | ConfigSidebarLink | ConfigSidebarPage,
+>(item: T, schema: any) {
+  const errors = [...Value.Errors(schema, item)]
+
+  // Add the item name for better errors
+  errors.forEach((error) => {
+    const errorMessage = `${error.message} on sidebar item: ${item.name}`
+    error.message = errorMessage
+  })
+  return errors
+}
```

Comment:
I think this is necessary because the full path for the object being validated is lost while iterating over the nested objects. Open to improvements on this